Starting having MainLoopTests.InvokeLeakTest failure after push develop branch to my fork .

```
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
  Failed Terminal.Gui.Core.MainLoopTests.InvokeLeakTest [1 s]
  Error Message:
[xUnit.net 00:01:04.81]     Terminal.Gui.Core.MainLoopTests.InvokeLeakTest [FAIL]
   System.TimeoutException : Timeout: Increment lost. tbCounter (41) didn't change after waiting 500 ms. Failed to reach 10000 on pass 1
  Stack Trace:
     at Terminal.Gui.Core.MainLoopTests.RunTest(Random r, TextField tf, Int32 numPasses, Int32 numIncrements, Int32 pollMs) in /home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/MainLoopTests.cs:line 556
   at Terminal.Gui.Core.MainLoopTests.<>c__DisplayClass24_0.<InvokeLeakTest>b__0() in /home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/MainLoopTests.cs:line 577
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Terminal.Gui.Core.MainLoopTests.InvokeLeakTest() in /home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/MainLoopTests.cs:line 582
--- End of stack trace from previous location ---

Attachments:
  /home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/TestResults/5774b41c-7d1a-461a-a399-f2edfe5328aa/coverage.opencover.xml
Failed!  - Failed:     1, Passed:   873, Skipped:     0, Total:   874, Duration: 1 m 9 s - /home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/bin/Debug/net6.0/UnitTests.dll (net6.0)
     4>Done Building Project "/home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/UnitTests.csproj" (VSTest target(s)) -- FAILED.
     1>Done Building Project "/home/runner/work/Terminal.Gui/Terminal.Gui/Terminal.sln" (VSTest target(s)) -- FAILED.

Build FAILED.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:01:19.40
Error: Process completed with exit code 1.

```

--- a/UnitTests/MainLoopTests.cs
+++ b/UnitTests/MainLoopTests.cs
@@ -527,14 +527,19 @@ namespace Terminal.Gui.Core {
 		// TODO: Add IMainLoop tests
 
 		volatile static int tbCounter = 0;
+		static ManualResetEventSlim _wakeUp = new ManualResetEventSlim (false);
 
-		private static void Launch (Random r, TextField tf)
+		private static void Launch (Random r, TextField tf, int target)
 		{
 			Task.Run (() => {
 				Thread.Sleep (r.Next (2, 4));
 				Application.MainLoop.Invoke (() => {
 					tf.Text = $"index{r.Next ()}";
 					Interlocked.Increment (ref tbCounter);
+					if (target == tbCounter) {
+						// On last increment wake up the check
+						_wakeUp.Set ();
+					}
 				});
 			});
 		}
@@ -542,16 +547,19 @@ namespace Terminal.Gui.Core {
 		private static void RunTest (Random r, TextField tf, int numPasses, int numIncrements, int pollMs)
 		{
 			for (int j = 0; j < numPasses; j++) {
+
+				_wakeUp.Reset ();
 				for (var i = 0; i < numIncrements; i++) {
-					Launch (r, tf);
+					Launch (r, tf, (j + 1) * numIncrements);
 				}
+
 
 				while (tbCounter != (j + 1) * numIncrements) // Wait for tbCounter to reach expected value
 				{
 					var tbNow = tbCounter;
-					Thread.Sleep (pollMs);
+					_wakeUp.Wait (pollMs);
 					if (tbCounter == tbNow) {
-						// No change after sleep: Idle handlers added via Application.MainLoop.Invoke have gone missing
+						// No change after wait: Idle handlers added via Application.MainLoop.Invoke have gone missing
 						Application.MainLoop.Invoke (() => Application.RequestStop ());
 						throw new TimeoutException (
 							$"Timeout: Increment lost. tbCounter ({tbCounter}) didn't " +
@@ -572,7 +580,7 @@ namespace Terminal.Gui.Core {
 
 			const int numPasses = 10;
 			const int numIncrements = 10000;
-			const int pollMs = 500;
+			const int pollMs = 20000;
 
 			var task = Task.Run (() => RunTest (r, tf, numPasses, numIncrements, pollMs));
 

